kii_cloud_api.go

```
@@ -172,25 +229,110 @@ func (au *APIAuthor) OnboardGateway(request OnboardGatewayRequest) (*OnboardGate
 		return nil, err
 	}
 	req.Header.Set("content-type", "application/vnd.kii.onboardingWithVendorThingIDByThing+json")
-	req.Header.Set("authorization", "bearer " + au.Token)
+	req.Header.Set("authorization", "bearer "+au.Token)
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	bodyStr, err := executeRequest(*req)
 	if err != nil {
 		return nil, err
+	} else {
+		err = json.Unmarshal(bodyStr, &ret)
+		if err != nil {
+			return nil, err
+		}
+		au.Token = ret.AccessToken
+		au.ID = ret.ThingID
+		return &ret, nil
 	}
-	defer resp.Body.Close()
+}
 
-	bodyStr, err := ioutil.ReadAll(resp.Body)
+// Request access token of end node of gateway.
+// When there's no error, GenerateEndNodeTokenResponse is returned.
+func (au APIAuthor) GenerateEndNodeToken(endnodeID string) (*GenerateEndNodeTokenResponse, error) {
+	var ret GenerateEndNodeTokenResponse
+	url := fmt.Sprintf("%s/things/%s/end-nodes/%s/token", au.App.KiiCloudBaseUrl(), au.ID, endnodeID)
+
+	reqObj := GenerateEndNodeTokenRequest{
+		ExpiresIn: "",
+	}
+	reqJson, _ := json.Marshal(reqObj)
```

lets not ignore error.